src/taipy/gui/page.py

```diff
@@ -85,10 +87,27 @@ def _get_content_detail(self, gui) -> str:
         return ""
 
     def _get_locals(self) -> t.Optional[t.Dict[str, t.Any]]:
-        return None if self._frame is None else _filter_locals(self._frame.f_locals)
+        return (
+            self._class_locals
+            if self._class_locals != {}
+            else None
+            if self._frame is None
+            else _filter_locals(self._frame.f_locals)
+        )
+
+    def _set_class_module(self, name: str, locals_: t.Dict[str, t.Any]) -> None:
+        self._class_module_name = name
+        self._class_locals = locals_
```

Comment:
What happens if class has no other member than page?

Reply:
then it should be an empty locals, as it should be